src/eventHandlers/WorkerInitHandler.ts

```diff
@@ -54,8 +63,34 @@ export class WorkerInitHandler extends EventHandler<'workerInitRequest', 'worker
         }
 
         logColdStartWarning(channel);
-        if (msg.functionAppDirectory) {
-            await channel.updatePackageJson(msg.functionAppDirectory);
+        const functionAppDirectory = nonNullProp(msg, 'functionAppDirectory');
+        await channel.updatePackageJson(functionAppDirectory);
+
+        const entryPointFile = channel.packageJson.main;
+        if (entryPointFile) {
+            channel.log({
+                message: `Loading entry point "${entryPointFile}"`,
+                level: LogLevel.Debug,
+                logCategory: LogCategory.System,
+            });
+            try {
+                const entryPointFullPath = path.join(functionAppDirectory, entryPointFile);
+                if (!(await pathExists(entryPointFullPath))) {
+                    throw new Error(`file does not exist`);
+                }
+
+                await loadScriptFile(entryPointFullPath, channel.packageJson);

```

Comment:
Loading a file in javascript means you're executing all the code at the top level of the file. That usually means importing modules, defining a class/function/const, and in our case registering a hook. As an example, 'hello' would get printed but 'world' would not:
```
console.log('hello');

export function world() {
    console.log('world');
}
```

That being said, users should put any complex code in a startup hook (which is the issue you're about to work on) instead of at the top level of their code. There's a few reasons for this that I already listed in https://github.com/Azure/azure-functions-nodejs-worker/issues/539

Reply:
Gotcha, that makes more sense now, thanks! 

Is "loading" standard terminology to refer to import/require? If not, renaming this to something like `executeScriptFile` would make it clearer to me what it's doing, but maybe that's just me :P

Reply:
Yeah I think loading is pretty standard. Its frowned upon to put complex code at the top level and complex code is the stuff you generally think of as "executing". If you take my examples from above, you could rephrase it to "loading" modules, "loading" classes/functions/consts and "loading" hooks. So while people _can_ execute complex stuff, they should ideally just be loading things